src/status_im2/common/floating_button_page/view.cljs

```diff
@@ -0,0 +1,93 @@
+           content-container-height header-height keyboard-shown?]}]
+  (let [available-space    (- window-height
+                              (safe-area/get-top)
+                              header-height
+                              keyboard-height ; Already contains the bottom safe area value
+                              footer-container-height)
+        scroll-view-height (- content-container-height content-scroll-y)
+        overlap?           (< available-space scroll-view-height)]
+    (and keyboard-shown? overlap?)))
+
+(defn- set-height-on-layout
+  [ratom]
+  (fn [event]
+    (let [height (oops/oget event "nativeEvent.layout.height")]
+      (reset! ratom height))))
+
+(defn view
+  [{:keys [header footer]} & children]
+  (reagent/with-let [window-height                (:height (rn/get-window))
+                     footer-container-height      (reagent/atom 0)
+                     header-height                (reagent/atom 0)
+                     content-container-height     (reagent/atom 0)
+                     content-scroll-y             (reagent/atom 0)
+                     keyboard-height              (reagent/atom 0)
+                     keyboard-will-show?          (reagent/atom false)
+                     keyboard-did-show?           (reagent/atom false)
+                     will-show-listener           (oops/ocall rn/keyboard
+                                                              "addListener"
+                                                              "keyboardWillShow"
+                                                              #(reset! keyboard-will-show? true))
+                     did-show-listener            (oops/ocall rn/keyboard
+                                                              "addListener"
+                                                              "keyboardDidShow"
+                                                              (fn [e]
+                                                                (reset! keyboard-height
+                                                                  (oops/oget e "endCoordinates.height"))
+                                                                (reset! keyboard-did-show? true)))
+                     will-hide-listener           (oops/ocall rn/keyboard
+                                                              "addListener"
+                                                              "keyboardWillHide"
+                                                              #(reset! keyboard-will-show? false))
+                     did-hide-listener            (oops/ocall rn/keyboard
+                                                              "addListener"
+                                                              "keyboardDidHide"
+                                                              #(reset! keyboard-did-show? false))
+                     set-header-height            (set-height-on-layout header-height)
```

Comment:
Nice work on setting up everything outside the renderer!

@J-Son89, @ulisesmac I think here we can learn from React hooks how to decouple views from stateful concepts, even if we're just using Reagent. The idea is simple, we extract to a function meaningful bundles of code and return a map of values/functions that operate as an interface of sorts that hides the inner complexities of the stateful code. I think this technique can be used much much more in our repo.

I suggest the following diff, which I very much enjoyed writing :) I wanted to see if it would be an improvement and if it actually works.

```diff
+(defn- init-keyboard-listeners
+  [{:keys [on-did-show]}]
+  (let [keyboard-will-show? (reagent/atom false)
+        keyboard-did-show?  (reagent/atom false)
+        will-show-listener  (oops/ocall rn/keyboard
+                                        :addListener
+                                        :keyboardWillShow
+                                        #(reset! keyboard-will-show? true))
+        did-show-listener   (oops/ocall rn/keyboard
+                                        :addListener
+                                        :keyboardDidShow
+                                        (fn [e]
+                                          (reset! keyboard-did-show? true)
+                                          (when on-did-show
+                                            (on-did-show e))))
+        will-hide-listener  (oops/ocall rn/keyboard
+                                        :addListener
+                                        :keyboardWillHide
+                                        #(reset! keyboard-will-show? false))
+        did-hide-listener   (oops/ocall rn/keyboard
+                                        :addListener
+                                        :keyboardDidHide
+                                        #(reset! keyboard-did-show? false))]
+    {:keyboard-will-show? keyboard-will-show?
+     :keyboard-did-show?  keyboard-did-show?
+     :remove-listeners    (fn []
+                            (doseq [listener [will-show-listener will-hide-listener did-show-listener
+                                              did-hide-listener]]
+                              (oops/ocall listener :remove)))}))
+
 (defn view
   [{:keys [header footer]} & children]
   (reagent/with-let [window-height                (:height (rn/get-window))
@@ -34,33 +64,19 @@
                      content-container-height     (reagent/atom 0)
                      content-scroll-y             (reagent/atom 0)
                      keyboard-height              (reagent/atom 0)
-                     keyboard-will-show?          (reagent/atom false)
-                     keyboard-did-show?           (reagent/atom false)
-                     will-show-listener           (oops/ocall rn/keyboard
-                                                              "addListener"
-                                                              "keyboardWillShow"
-                                                              #(reset! keyboard-will-show? true))
-                     did-show-listener            (oops/ocall rn/keyboard
-                                                              "addListener"
-                                                              "keyboardDidShow"
-                                                              (fn [e]
-                                                                (reset! keyboard-height
-                                                                  (oops/oget e "endCoordinates.height"))
-                                                                (reset! keyboard-did-show? true)))
-                     will-hide-listener           (oops/ocall rn/keyboard
-                                                              "addListener"
-                                                              "keyboardWillHide"
-                                                              #(reset! keyboard-will-show? false))
-                     did-hide-listener            (oops/ocall rn/keyboard
-                                                              "addListener"
-                                                              "keyboardDidHide"
-                                                              #(reset! keyboard-did-show? false))
+                     {:keys [keyboard-will-show?
+                             keyboard-did-show?
+                             remove-listeners]}   (init-keyboard-listeners
+                                                   {:on-did-show
+                                                    (fn [e]
+                                                      (reset! keyboard-height
+                                                        (oops/oget e :endCoordinates.height)))})
                      set-header-height            (set-height-on-layout header-height)
                      set-content-container-height (set-height-on-layout content-container-height)
                      set-footer-container-height  (set-height-on-layout footer-container-height)
                      set-content-y-scroll         (fn [event]
                                                     (reset! content-scroll-y
-                                                      (oops/oget event "nativeEvent.contentOffset.y")))]
+                                                      (oops/oget event :nativeEvent.contentOffset.y)))]
     (let [keyboard-shown?  (if platform/ios? @keyboard-will-show? @keyboard-did-show?)
           show-background? (show-background {:window-height            window-height
                                              :footer-container-height  @footer-container-height
@@ -89,5 +105,4 @@
           :blur?           show-background?}
          footer]]])
     (finally
-     (doseq [listener [will-show-listener will-hide-listener did-show-listener did-hide-listener]]
-       (oops/ocall listener "remove")))))
+     (remove-listeners))))
```

Reply:
nice suggestion! :)

Reply:
Thanks @ilmotta it sounds good :+1: 
the reading of this component is hard, it has been definitely improved :+1: